src/app/fdctl/run/tiles/fd_gui.c

```diff
@@ -270,6 +288,51 @@ privileged_init( fd_topo_t *      topo,
   ctx->identity_key = fd_keyload_load( tile->gui.identity_key_path, /* pubkey only: */ 1 );
 }
 
+/* pre_compress_files compresses static assets into wksp-provided
+   buffers using Zstandard.  Logs warning and continues if insufficient
+   wksp space is available. */
+
+static void
+pre_compress_files( fd_wksp_t * wksp ) {
+# if !FD_HAS_ZSTD
+  (void)wksp: (void)files;
+  return;
+# else
+
+  ulong uncompressed_sz = dist_file_sz();
+  ulong glo, ghi;
+  if( FD_UNLIKELY( !fd_wksp_alloc_at_least( wksp, FD_SPAD_ALIGN, uncompressed_sz/2UL, 1UL, &glo, &ghi ) ) ) {
+    FD_LOG_WARNING(( "Failed to allocate space for compressing assets" ));
+    return;
+  }
+  fd_spad_t * spad = fd_spad_join( fd_spad_new( fd_wksp_laddr_fast( wksp, glo ), fd_spad_mem_max_max( ghi-glo ) ) );
+  fd_spad_push( spad );
+  for( fd_http_static_file_t * f=STATIC_FILES; f->name; f++ ) {
+    char const * ext = strrchr( f->name, '.' );
+    if( !ext ) continue;
+    if( !strcmp( ext, ".html" ) ||
+        !strcmp( ext, ".css"  ) ||
+        !strcmp( ext, ".js"   ) ||
+        !strcmp( ext, ".svg"  ) ) {}
+    else continue;
+
+    ulong   zstd_bufsz = fd_spad_alloc_max( spad, 1UL );
+    uchar * zstd_buf   = fd_spad_prepare( spad, 1UL, zstd_bufsz );
+    ulong zstd_sz = ZSTD_compress( zstd_buf, zstd_bufsz, f->data, *f->data_len, 19 );
```

Comment:
This seems to do malloc() under the hood. Need to switch to the static compression API

Reply:
Fixed